XenAdmin/SettingsPanels/GpuEditPage.cs

```diff
@@ -292,7 +292,9 @@ private void addButton_Click(object sender, EventArgs e)
                         return;
 
                     VGPU_type type = tuple.VgpuTypes[0];
-                    var vGpu = new VGPU();
+                    // Set vGPU device to VGPU_UNASSIGNED_DEVICE, does not assign any devcie ID to it.
+                    // When create vGPU, 0 is used to declare the first availabe slot in VM
+                    var vGpu = new VGPU (GpuAssignAction.VGPU_UNASSIGNED_DEVICE);

```

Comment:
Since the constructors of the API classes are autogenerated and a new one cannot be added manually, this should be instead
```
var vGpu = new VGPU();
vGpu.device = GpuAssignAction.VGPU_UNASSIGNED_DEVICE;
```